verrazzano/verrazzano
# VZ-11326 Move all cluster management to VMC controller

Cluster operator controllers have been refactored as follows:
- CAPI controller is mostly responsible for creating a VMC when a CAPI cluster is created and removing it when it is deleted
- The VMC controller now handles the management and pushing of cluster related resources via Rancher or CAPI mechanisms depending on whether CAPI cluster creation, Rancher is enabled, VZ is installed etc.

Testing done:
- VZ, CAPI, and MC pipelines built
- Manual scenarios:
1.  Check VMC and Rancher status for a CAPI cluster creation, followed by installation of VZ, followed by enabling Rancher
2. Check VMC and Rancher status for CAPI cluster creation with Rancher enabled, followed by installation of VZ
3. Rancher UI generated cluster, w/o VZ followed by enabling VZ (in progress)
